Production/govuk_ios/Extensions/Foundation/Codable+Extensions.swift

```diff
@@ -0,0 +1,40 @@
+import Foundation
+
+extension Decodable {
+    func extractDataSliceForKey(_ key: CodingKey,
```

Comment:
Wondering if this function body can be broken down a bit into private function(s) to aid readability?

Reply:
Done